simple_kp/build_db.py

```diff
@@ -26,40 +26,47 @@ async def get_data_from_string(data: str):
         r"(?P<o2s><?)-- predicate (?P<predicate>[\w:]+) --(?P<s2o>>?)"
         r"(?P<target>[\w:]+)"
     )
-    nodes = []
-    edges = []
-    for idx, line in enumerate(data.split("\n")):
+    nodes = {}
+    edges = {}
+    for line in data.split("\n"):
         line = line.strip()
         if not line:
             continue
+
         match = re.fullmatch(node_pattern, line)
         if match is not None:
-            nodes.append({
-                "id": match.group("id"),
-                "category": match.group("category"),
-            })
+            nid = match.group("id")
+            if nid not in nodes:
+                nodes[nid] = {
+                    "id": nid,
+                    "category": [],
+                }
+
+            nodes[nid]["category"].append(
+                match.group("category")
+            )
             continue
+
         match = re.fullmatch(edge_pattern, line)
         if match is not None:
+            eid = f"{match.group('source')}-{match.group('target')}"
```

Comment:
There may be multiple edges between the same pair of nodes. I'd just generate a random string for edge ids.